src/System.Management.Automation/DscSupport/JsonDscClassCache.cs

```diff
@@ -0,0 +1,3249 @@
+
+                var metaConfigExtensionFile = Path.Join(systemResourceRoot, "BaseRegistration", "MSFT_MetaConfigurationExtensionClasses.schema.json");
+                ImportClasses(metaConfigExtensionFile, DefaultModuleInfoForMetaConfigResource, errors);
+
+                // Load all of the system resource schema files, searching
+                string resources;
+                foreach (var resourceRoot in allResourceRoots)
+                {
+                    resources = Path.Join(resourceRoot, "Schema");
+                    if (!Directory.Exists(resources))
+                    {
+                        continue;
+                    }
+
+                    foreach (var schemaFile in Directory.EnumerateFiles(resources, "*.schema.json", SearchOption.AllDirectories))
+                    {
+                        ImportClasses(schemaFile, s_defaultModuleInfoForResource, errors);
+                    }
+                }
+
+                // Load Regular and DSC PS modules
+                bool importInBoxResourcesImplicitly = false;
+                List<string> modulePaths = new List<string>();
+                if (modulePathList?.Count == 0)
+                {
+                    modulePaths.Add(Path.Join(configSystemPath, inboxModulePath));
+                    importInBoxResourcesImplicitly = true;
+                }
+                else
+                {
+                    if (modulePathList != null)
+                    {
+                        foreach (string moduleFolderPath in modulePathList)
+                        {
+                            if (!Directory.Exists(moduleFolderPath))
+                            {
+                                continue;
+                            }
+
+                            foreach (string moduleDir in Directory.EnumerateDirectories(moduleFolderPath))
+                            {
+                                modulePaths.Add(moduleDir);
+                            }
+                        }
+                    }
+                }
+
+                LoadDSCResourceIntoCache(errors, modulePaths, importInBoxResourcesImplicitly);
+            }
+        }
+
+        /// <summary>
+        /// Load DSC resources into Cache from moduleFolderPath.
+        /// </summary>
+        /// <param name="errors">Collection of any errors encountered during initialization.</param>
+        /// <param name="modulePathList">Module path from where DSC PS modules will be loaded.</param>
+        /// <param name="importInBoxResourcesImplicitly">
+        /// if module is inbox.
+        /// </param>
+        private static void LoadDSCResourceIntoCache(Collection<Exception> errors, List<string> modulePathList, bool importInBoxResourcesImplicitly)
+        {
+            foreach (string moduleDir in modulePathList)
+            {
+                if (!Directory.Exists(moduleDir)) continue;
+
+                var dscResourcesPath = Path.Join(moduleDir, "DscResources");
```

Comment:
This method doesn't cover the case where v3 `PSDesiredStateConfiguration` module can be installed with other version side by side. If that can happen, then you won't find a `DscResources` directly under `PSDesiredStateConfiguration` folder.